tests/trainer/test_fsdp_checkpoint.py

```diff
@@ -767,6 +776,17 @@ def test_fsdp_partitioned_state_dict_load(
     save_filename = 'ba{batch}-rank{rank}.pt'
 
     fsdp_config = FSDPConfig(state_dict_type='sharded')
+    tp_config = None
+    if use_tp:
+        fsdp_config = FSDPConfig(data_parallel_shard_degree=1)
```

Comment:
IIUC, this is essentially TP only test? Can we also add HSDP + TP, and FSDP + TP? I know we don't support multi-gpu auto-test, but maybe at least add those and run locally for now ?